packages/dev/core/src/scene.ts

```diff
@@ -514,9 +515,44 @@ export class Scene implements IAnimatable, IClipPlanesHolder, IAssetContainer {
         }
 
         this._environmentTexture = value;
+        if (this._importanceSamplingRenderer && value) {
+            this._importanceSamplingRenderer.iblSource = value;
+        }
         this.markAllMaterialsAsDirty(Constants.MATERIAL_TextureDirtyFlag);
     }
 
+    private _useEnvironmentCDFMaps = false;
+
+    /**
+     * Specifies whether IBL filtering will use the CDF (cumulative distribution function)
+     * maps for importance sampling.
+     */
+    public get useEnvironmentCDFMaps(): boolean {

```

Comment:
This creates a dependency of the scene to  the importanceSamplingRenderer which we should avoid for better tree shaking. Either you need an async function to turn it on with dynamic import or only store the property on the scene and user will need to do the new ?

Reply:
What about creating a scene component like the geometryBufferRenderer is doing?

Reply:
I do not think you need to hook in the same kind of functions, but you could extend the scene prototype in the same way in your renderer. The only drawback is that it works with side effects and we are trying to avoid them.

cc @RaananW if he has a better idea to not get in the way of removing side effects ?

Reply:
I would suggest to simply move it out of the scene, but I see that it is also needed when the environment texture is changed.
A scene component will be the better option for the time being. When I get to making the esm changes I will also work on this component, so feel free to continue using the same methods we used so far.

Reply:
Thank you. I've done this now.
To replace the functionality that I had previously added to the `scene.environmentTexture` setter, I've added a new observable for when the environmentTexture changes. This way, the CDF maps can be regenerated whenever this changes. Is this an okay way to handle things?